test/github-issues/9229/entity/test.entity.ts

```diff
@@ -0,0 +1,17 @@
+import {
+    Column,
+    Entity,
+    JoinColumn,
+    OneToOne,
+    PrimaryColumn,
+} from "../../../../src"
+@Entity("testEntity")
+export class TestEntity {
+    @PrimaryColumn()
+    @OneToOne("User", "id")
+    @JoinColumn({ name: "userId" })
+    public userId: string
```

Comment:
I just left a comment in #9229. It has invalid decorators usage. Same issue in your test.

Use case is not valid. Relational decorators must be applied to the properties with the type of the object to which relation maps, e.g.:

```ts
import { Column, Entity, JoinColumn, OneToOne, PrimaryColumn } from 'typeorm';
@Entity('testEntity')
export class TestEntity {
    @PrimaryColumn()
    public userId: string;

    @OneToOne('User', 'test')
    @JoinColumn({ name: 'userId' })
    public user: User;

    @Column({ type: 'jsonb', nullable: false })
    public x: string[];
}
```

Change the test, and run the test without your changes in the `RelationJoinColumnBuilder.ts` file.

Reply:
Yeah actually I know too this use case is not valid. But when user added both PrimaryColumn and JoinColumn accidentally like this example,  our query builder generates table query as below. 

```
CREATE TABLE "testEntity" (
    "userId" uuid NOT NULL,
    "x" jsonb NOT NULL,
    CONSTRAINT "REL_3138077fc5fcc8a4e4e3c175f4" UNIQUE ("userId"),
    CONSTRAINT "PK_3138077fc5fcc8a4e4e3c175f40" PRIMARY KEY ("userId")
)
```

I manually ran this query and similar queries which includes both primary and unique constraints on same field in our db and I saw postgresql added only the last constraint. My code includes the change that only prevent this condition.

Reply:
we shouldn't bring fixes for invalid use cases. Other parts of the codebase might not expect invalid use cases, thus they can break or lead to incorrect behavior. We need to update the test cases to valid code and check if we still have this issue with a valid entity definitions.

Reply:
> Relational decorators must be applied to the properties with the type of the object to which relation maps

It may be useful to do some kind assertion there.